fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java

```diff
@@ -655,6 +666,224 @@ private void parse() throws AnalysisException, DdlException {
+                    for (Expr expr : predicate.getChildren()) {
+                        SlotRef slotRef = (SlotRef) expr;
+                        if (tupleIds.contains(slotRef.getDesc().getParent().getId())) {
+                            tmpRef = slotRef;
+                        } else {
+                            outerRef = slotRef;
+                            ++matchCount;
+                        }
+                        if (outerRef != null && innerRef == null) {
+                            innerRef = tmpRef;
+                        }
+                    }
+                }
+            }
+        }
+
+        if (matchCount == 1) {
+            // for simplicity, we only handle the case that:
+            // 1. only 1 outer slot ref is used in scalar subquery's correlatedPredicates's operator
+            // 2. there is no same tableRef in subquery as outerRef's tableRef
+            List<TableRef> tableRefs = selectStmt.getTableRefs();
+            long outerTableId = outerRef.getTable().getId();
+            for (TableRef tableRef : tableRefs) {
+                if (tableRef.getTable().getId() == outerTableId) {
+                    return null;
+                }
+            }
+
+            // make sure outer table have alias
+            TupleDescriptor outerTupleDescriptor = outerRef.getDesc().getParent();
+            if (!outerTupleDescriptor.hasExplicitAlias()) {
+                String alias = UUID.randomUUID().toString();
+                // set outer table's alias
+                outerTupleDescriptor.getRef().setAlias(alias);
+                // set the tuple descriptor's alias
+                outerTupleDescriptor.setAliases(new String[]{alias}, true);
+                // update tuple descriptor's alias in parent analyzer
+                selectStmt.getAnalyzer().getParentAnalyzer().setTupleDescriptorAlias(outerTupleDescriptor, alias);
+            }
+
+            // make sure all outer table slots' have an alias table name
+            List<SlotRef> slotRefs = Lists.newArrayList();
+            Expr.collectList(conjuncts, SlotRef.class, slotRefs);
+            TupleDescriptor slotTupleDescriptor;
+            for (SlotRef slotRef : slotRefs) {
+                slotTupleDescriptor = slotRef.getDesc().getParent();
+                if (slotTupleDescriptor.getId() == outerTupleDescriptor.getId() && slotRef.getOriginTableName() == null) {
+                    slotRef.setTblName(new TableName(null, outerTupleDescriptor.getAlias()));
+                }
+            }
+
+            return new Pair<>(outerRef, innerRef);
+        } else {
+            return null;
+        }
+    }
+
+    private List<Subquery> findSubqueryInConjuncts(List<Expr> conjuncts) {
+        List<Subquery> subqueries = Lists.newArrayList();
+        for (Expr conj : conjuncts) {
+            if (conj instanceof Predicate) {
+                Predicate predicate = (Predicate) conj;
+                for (Expr child : predicate.getChildren()) {
```

Comment:
The predicate can be `Or Predicate`.
Because line 728 `.getConjuncts()` just split the `And Predicate`